Remove compilers warnings due to set_mb by fixing macro.

Signed-off-by: Cyril Dailly <cyril.dailly@st.com>
Index: linux/include/asm-sh/system.h
===================================================================
--- linux.orig/include/asm-sh/system.h	2006-12-14 15:17:34.622686000 +0000
+++ linux/include/asm-sh/system.h	2007-02-13 08:20:22.101842000 +0000
@@ -137,7 +137,7 @@
 #define smp_read_barrier_depends()	do { } while(0)
 #endif
 
-#define set_mb(var, value) do { xchg(&var, value); } while (0)
+#define set_mb(var, value) do { (void)xchg(&var, value); } while (0)
 #define set_wmb(var, value) do { var = value; wmb(); } while (0)
 
 /* Interrupt Control */
